### PR TITLE
Extract refactoring

### PR DESCRIPTION
**Pull request by MFlisar**
Fixed:
* SearchBar supports user date format and format is based and formatter with SimpleDateFormat now
* fixes widget drag & drop because of DragDropHandler introduction
* fixed widget drag & drop behaviour => was jumping around when starting drag & drop because touch listener was not saving the correct touch point
* added the ability to disable the date/time display in the search bar and also avoid the unnecessary broadcast receiver in this case
* DragDropHandler uses new drag function on android >= 24
* WidgetView touch handling improved
* moved LauncherLoadingIcon away from the core to the app
* cleaned resources, only some of them are still in the core; no duplicate strings anymore
* fixed animation of searchbar when filtering
* less restrictive setup class

I just looked through the issues, following 3 should be solved by my commits or the last ones. Only the setup for the custom date format is missing in the UI, technically custom date formats are fully supported now.

https://github.com/OpenLauncherTeam/openlauncher/issues/141
https://github.com/OpenLauncherTeam/openlauncher/issues/115
https://github.com/OpenLauncherTeam/openlauncher/issues/112

### CODE REVIEW

File: core/src/main/java/com/benny/openlauncher/core/activity/Home.java [hunk 1]
```diff
@@ -125,6 +111,18 @@ protected void onCreate(Bundle savedInstanceState) {
 
         if (!Setup.wasInitialised())
             initStaticHelper();
+
+        if (Setup.appSettings().searchBarTimeEnabled()) {
+            timeChangedReceiver = new BroadcastReceiver() {
```
Comment (gsantner) [inline core/src/main/java/com/benny/openlauncher/core/activity/Home.java]: not unregistered

File: core/src/main/java/com/benny/openlauncher/core/model/BaseIconLabelItem.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import android.support.v7.widget.RecyclerView;
+import android.view.View;
+
+import com.benny.openlauncher.core.R;
+import com.benny.openlauncher.core.interfaces.FastItem;
+import com.benny.openlauncher.core.interfaces.LabelProvider;
+import com.mikepenz.fastadapter.items.AbstractItem;
+
+/**
+ * Created by flisar on 28.06.2017.
```
Comment (gsantner) [inline core/src/main/java/com/benny/openlauncher/core/model/BaseIconLabelItem.java]: \-

File: core/src/main/java/com/benny/openlauncher/core/activity/Home.java (hunk 1 above)
Comment (MFlisar) [inline core/src/main/java/com/benny/openlauncher/core/activity/Home.java]: weird, don't know how this could happen...

Comment (MFlisar) [inline core/src/main/java/com/benny/openlauncher/core/activity/Home.java]: Now I know, it has never been unregistered before ;-)

**Comment by gsantner** [comment]
@MFlisar Is this otherwise mergeable? So we don't diverge too much when fixing bugs and the `split up`-branch cannot merge it back

**Pull request merged**